### PR TITLE
[Filing app] Phase 2 - Live data

### LINKED ISSUE #285: [Task] Filing app - Skeleton page (Phase 1 - mock data)
- [x] Launch the filing platform - https://github.com/cfpb/sbl-project/issues/77
- [x] See list of active institutions I can file for - https://github.com/cfpb/sbl-project/issues/120 (Phase 1)
  - [x] See current Filing status of each institution - https://github.com/cfpb/sbl-project/issues/121 (Phase 1)
  - [x] Start the filing process for an associated institution - https://github.com/cfpb/sbl-project/issues/123 (Phase 1)

### LINKED ISSUE #286: [Task] Filing app - Skeleton page (Phase 2 - live data)
- [x] See list of active institutions I can file for - https://github.com/cfpb/sbl-project/issues/120 (Phase 2)
  - [x] See current Filing status of each institution - https://github.com/cfpb/sbl-project/issues/121 (Phase 2)
  - [x] Start the filing process for an associated institution - https://github.com/cfpb/sbl-project/issues/123 (Phase 2)

Notes
- [Filing API endpoints](https://github.com/cfpb/sbl-filing-api/wiki/Endpoints-%E2%80%90-Filings)
- [Sequence diagrams](https://github.com/cfpb/sbl-filing-api/issues/34)

### PR DESCRIPTION
**Pull request by meissadia**
Closes #286 

## Changes

- Pulls associated institutions from API
- Enables creation of a Filing from the UI (via the API)
- Displays current Filing status, as best we can determine, from API data

## How to test this PR

1. Log into the Frontend with a user that has at least one associated institution
2. Should see a status message indicating: `You have not started the Filing process`
3. Click `Start a Filing`
4. Should see a status message: `Upload your lending data`

## Screenshots
- Implemented statuses: `Start a Filing`, `Upload your data`
  Screenshot 2024-03-12 at 1 35 06 PM
- Display when there's an error fetching data
  Screenshot 2024-03-12 at 1 14 06 PM

### CODE REVIEW

**Comment by billhimmelsbach** [comment]
Heyo! Can I review this @meissadia with the assumption that the other PR will get merged?

**Comment by meissadia** [comment]
> Heyo! Can I review this @meissadia with the assumption that the other PR will get merged?

@billhimmelsbach Yes, integration is working well, left PR in draft to allow prerequisites to get reviewed/merged.

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
Heyo! This is looking great! There's a different endpoint that we want to use though to see if someone has started a filing for a given LEI.

Instead of `Get Submission Latest` (`/v1/filing/institutions/{lei}/filings/{period_name}/submissions/latest`), we'll want to use `Get Filing` (`/v1/filing/institutions/{lei}/filings/{period_name}`).

We'll be using the `Get Submission Latest` endpoint later when we're trying to get the most recent upload/submission attempt of SBLAR files. If you use the `Get Filing` api endpoint, you can look for a `204` instead of a `500`, which will help with error handling.

Screenshot 2024-03-20 at 3 44 59 PM

**Comment by meissadia** [comment]
Implemented a working solution with the `/v1/filing/institutions/{lei}/filings/{period_name}` endpoint but still need to address the other refactorings.

File: src/utils/fetchFilingSubmissionLatest.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { request } from 'api/axiosService';
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/utils/fetchFilingSubmissionLatest.tsx:1]: Moved 👍🏾

File: src/pages/Filing/FilingApp/InstitutionCard.tsx [hunk 2]
```diff
@@ -38,23 +42,31 @@ function SecondaryButton({
...
+  let status = '';
+
+  if (isLoading)
     return (
       <div>
         <Icon name='updating' /> Loading submission status...
       </div>
     );
 
+  if (axios.isAxiosError(error) && error.response?.status === Error500)
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/Filing/FilingApp/InstitutionCard.tsx]: Updated to use requested endpoint.  
 
The way our [axiosService](https://github.com/cfpb/sbl-frontend/blob/main/src/api/axiosService.ts#L31-L46) is currently setup, the `request()` function does not allow access to the API requests' status.  We may need to refactor this to be able to make more refined Frontend decisions using those status codes.

**Comment by meissadia** [comment]
Adjustments complete, ready for re-review.

**Comment by billhimmelsbach** [comment]
Not sure what's wrong with my setup, but I get this error consistently, at least on the latest for the repos.

Screenshot 2024-03-25 at 6 48 05 PM

Seems like the auth token is only good for the first API call?, even when I'm just using bruno. I'll schedule a meeting after stand-up and you can walk me through how it's looking for you?

**Review by billhimmelsbach — APPROVED** [review]
Heyo! Thanks for debugging it with me, I'll push up a PR with a fix after this. For now, this looks great! 🎉

**Pull request merged**